p2panda-stream/src/ordering/partial/store.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+use std::hash::Hash as StdHash;
+
+use super::PartialOrderError;
+
+/// Trait defining a store API for handling ready and pending dependencies.
+///
+/// An implementation of this store trait provides the following functionality:
+/// - maintain a list of all items which have all their dependencies met
+/// - maintain a list of items which don't have their dependencies met
+/// - return all pending items which depend on a given item
+pub trait PartialOrderStore<K>
+where
+    K: Clone + Copy + StdHash + PartialEq + Eq,
+{
+    /// Add an item to the store which has all it's dependencies met already.
+    async fn add_ready(&mut self, key: K) -> Result<bool, PartialOrderError>;
```

Comment:
Just a thought: How about `mark_ready` and `mark_pending`?

Reply:
Yeh, I like that!